### ISSUE TITLE
Cannot signin using google signin

### ISSUE THREAD

**Issue by kingmaker000** [flutter/flutter#97043]
I have used this code in past without any platform exception. But now i am facing platform exception.

```
`
  final googlesignin= GoogleSignIn(scopes: [
    'email',
    'https://www.googleapis.com/auth/contacts.readonly',
  ],);

  GoogleSignInAccount? _user;

  GoogleSignInAccount get user => _user!;

 void googleSignIn() async{
    setState(() {
      isLoading=true;
    });
    final googleUser = await googlesignin.signIn();

    if(googleUser == null) return;

    _user=googleUser;

    final googleAuth= await googleUser.authentication;
    final credential= GoogleAuthProvider.credential(accessToken: googleAuth.accessToken,idToken: googleAuth.idToken);

    final authresult = await FirebaseAuth.instance.signInWithCredential(credential);

    await  FirebaseFirestore.instance.collection('users').doc(authresult.user!.uid).set(
        {'email': _user!.email,'name': _user!.displayName,'contact': '','imageUrl': _user!.photoUrl,'address': '','latitude': null,'longitude': null});
    await FirebaseFirestore.instance.collection('timings').doc(authresult.user!.uid).collection('logintime').add({'time': DateTime.now()});
  }`
```

**Comment by cedvdb** [comment]
This is the wrong repository

**Comment by darshankawar** [comment]
@kingmaker000 
Can you provide the complete error log that shows the platformException ?

**Comment by kingmaker000** [comment]
When i try google sign in this was the error,

`
I/ProviderInstaller(19649): Installed default security provider GmsCore_OpenSSL
D/EGL_emulation(19649): app_time_stats: avg=433.24ms min=17.30ms max=4060.73ms count=10
D/EGL_emulation(19649): app_time_stats: avg=884.02ms min=90.04ms max=3127.30ms count=4
D/EGL_emulation(19649): app_time_stats: avg=9271.90ms min=2.71ms max=203867.44ms count=22
E/flutter (19649): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: PlatformException(sign_in_failed, com.google.android.gms.common.api.ApiException: 12500: , null, null)
E/flutter (19649): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:607:7)
E/flutter (19649): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:167:18)
E/flutter (19649): <asynchronous suspension>
E/flutter (19649): #2      MethodChannel.invokeMapMethod (package:flutter/src/services/platform_channel.dart:367:43)
E/flutter (19649): <asynchronous suspension>
E/flutter (19649): #3      GoogleSignIn._callMethod (package:google_sign_in/google_sign_in.dart:244:30)
E/flutter (19649): <asynchronous suspension>
E/flutter (19649): #4      GoogleSignIn.signIn.isCanceled (package:google_sign_in/google_sign_in.dart)
E/flutter (19649): <asynchronous suspension>
E/flutter (19649): 
D/EGL_emulation(19649): app_time_stats: avg=23.14ms min=3.73ms max=123.75ms count=43
D/EGL_emulation(19649): app_time_stats: avg=7.71ms min=2.98ms max=26.32ms count=59
D/EGL_emulation(19649): app_time_stats: avg=5.20ms min=3.15ms max=8.94ms count=60
D/EGL_emulation(19649): app_time_stats: avg=5.83ms min=3.62ms max=13.34ms count=60
`

**Comment by kingmaker000** [comment]
When i try to use google sign in this was the exception raised,

I/ProviderInstaller(19649): Installed default security provider GmsCore_OpenSSL
D/EGL_emulation(19649): app_time_stats: avg=433.24ms min=17.30ms max=4060.73ms count=10
D/EGL_emulation(19649): app_time_stats: avg=884.02ms min=90.04ms max=3127.30ms count=4
D/EGL_emulation(19649): app_time_stats: avg=9271.90ms min=2.71ms max=203867.44ms count=22
E/flutter (19649): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: PlatformException(sign_in_failed, com.google.android.gms.common.api.ApiException: 12500: , null, null)
E/flutter (19649): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:607:7)
E/flutter (19649): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:167:18)
E/flutter (19649): <asynchronous suspension>
E/flutter (19649): #2      MethodChannel.invokeMapMethod (package:flutter/src/services/platform_channel.dart:367:43)
E/flutter (19649): <asynchronous suspension>
E/flutter (19649): #3      GoogleSignIn._callMethod (package:google_sign_in/google_sign_in.dart:244:30)
E/flutter (19649): <asynchronous suspension>
E/flutter (19649): #4      GoogleSignIn.signIn.isCanceled (package:google_sign_in/google_sign_in.dart)
E/flutter (19649): <asynchronous suspension>
E/flutter (19649): 
D/EGL_emulation(19649): app_time_stats: avg=23.14ms min=3.73ms max=123.75ms count=43
D/EGL_emulation(19649): app_time_stats: avg=7.71ms min=2.98ms max=26.32ms count=59
D/EGL_emulation(19649): app_time_stats: avg=5.20ms min=3.15ms max=8.94ms count=60
D/EGL_emulation(19649): app_time_stats: avg=5.83ms min=3.62ms max=13.34ms count=60

**Comment by darshankawar** [comment]
@kingmaker000 
Check below links and see if they help:

https://stackoverflow.com/questions/56188338/platformexception-platformexceptionsign-in-failed-com-google-android-gms-comm

https://stackoverflow.com/questions/53094767/exception-com-google-android-gms-common-api-apiexception-12500

[#37242 (comment)](https://github.com/flutter/flutter/issues/37242#issuecomment-819384488)

https://stackoverflow.com/questions/47632035/google-sign-in-error-12500

**Comment by kingmaker000** [comment]
Thank you for your help.

https://developers.google.com/android/guides/client-auth

I have solved this issue by following this link. I had been using SHA 1 and SHA 256 key of release version (get this key using "gradlew signingReport" command) in the Firebase console finger print. But now I used the command in the above link and i got the debug SHA1 and SHA256 key .we can also find this key using the command "gradlew singningReport"(Use the debug key not release key).

**Comment by kingmaker000** [comment]
Thankyou for your help. I have tried the solutions in those links.  

Now I changed the SHA key of release version to debug version. I got this keys by following this link.
https://developers.google.com/android/guides/client-auth

And now I am getting an error like "Google does not verified this app" when I try to login with google account. Below this error dialogue box there is one option called advanced. When I choose this I can login. This happens only when I am running my app in android studio. But after debugging or publishing app, this option wont appear.

In google cloud platform I have submitted my app for verification. But verification not yet completed. Do we need to verify our app in GCP?  

- [ ] 
<img width="540" alt="Screenshot_1642768621" src="https://user-images.githubusercontent.com/86338457/151912441-54b05c25-a1ea-4b49-90fc-613ba4214ecb.png">
